### DISCUSSION TITLE
Decoding works in the decoding script, but the routing tab doesn't provide output

### DISCUSSION THREAD

**Discussion by marjgh** [opendcs/opendcs#446] [Q&A]
Hi, i am trying to use a csv file as my input to the decoding, when i decode the raw message in decoding scripts/under configs tab, it shows my decoded data in the table, but when i try to use routing tab it doesn't show any output (after i run schedule entry). what could be the reason?

![2024-01-11_14-40-04](https://github.com/opendcs/opendcs/assets/124637860/b22b8ba1-83f1-417b-b3e7-94eda5200019)
![2024-01-11_14-41-42](https://github.com/opendcs/opendcs/assets/124637860/fad74038-b370-40e0-aa89-b2ab14160ae7)
![2024-01-11_14-42-24](https://github.com/opendcs/opendcs/assets/124637860/8a520508-a206-4e84-846b-7f7cac088390)

**Comment by MikeNeilson** [comment]
Is the routing scheduler running?

I'll have to check but I don't believe the run once option here actually runs the routingspec; pretty sure there's an issue open requesting a button that does so.

**Reply by marjgh** [reply]
when I check the routing spec it shown no output, and it sounds like it is not running, but I am using the same routing scheduler i use for other data, like WSC Datamart, and it works perfectly for them. 
![2024-01-11_14-58-45](https://github.com/opendcs/opendcs/assets/124637860/d29bbca6-a3ac-4484-8482-06d3a2fddc2e)

**Comment by MikeNeilson** [comment]
Can you search the log for the RoutingSpec name and the Config name?
Given that the decoding it self works my next guess would be something in the config that's causing to just not process the required data at all, like a typo in the source directory name or the like.

**Reply by marjgh** [reply]
well, i checked the source and they are all correct. is there any other probable reasons?
![1](https://github.com/opendcs/opendcs/assets/124637860/2a623042-2f59-41f2-b7e2-daf0ca7fa181)
![2](https://github.com/opendcs/opendcs/assets/124637860/bdeceb6b-0314-45c3-a8bd-1981476fb8d4)
![3](https://github.com/opendcs/opendcs/assets/124637860/a9e83d04-257c-48b8-a503-16d4ba4d5004)
![4](https://github.com/opendcs/opendcs/assets/124637860/ffced884-698d-4b97-9726-349d3d1a5ef0)
![5](https://github.com/opendcs/opendcs/assets/124637860/d95d4960-e21f-42e7-a284-d4b55e1c20ae)
![6](https://github.com/opendcs/opendcs/assets/124637860/9c9af5e2-333d-4ac7-b537-50d7d1101e39)
![7](https://github.com/opendcs/opendcs/assets/124637860/0878c680-b8fe-4e6c-91bd-d70ba7411db6)

**Reply by MikeNeilson** [reply]
Many, but to properly diagnose and search the possibilities the logs are required.
The instance of the RoutingScheduler that is running should have a log.

**Reply by marjgh** [reply]
thanks, and where/how can i check the log?

**Reply by MikeNeilson** [reply]
That depends on your installation. If you, or whomever administers the systems, didn't explicitly set a log directory they'll be in the working directory from which the commands were started.

**Reply by marjgh** [reply]
Ok I think i found that, and as you said when i run it in schedule entry, it sounds like it is not running.
![8](https://github.com/opendcs/opendcs/assets/124637860/33c2fe26-0f70-473e-ab77-81a3bb2dd0ce)
![9](https://github.com/opendcs/opendcs/assets/124637860/3a2bbeb4-7ab8-4666-8656-62d08d64e393)

**Reply by marjgh** [reply]
![10](https://github.com/opendcs/opendcs/assets/124637860/0c069da9-9f5e-45d0-ae7f-f8d5c64dc952)

**Comment by agilmore2** [comment]
Looks to me like there's a germane error about failing to parse a GOES DCS
header from the file. I believe the source may need some more configuration
to tell it what type of header to read (may be other/none)

The decoding bit looks to be a little off as well, it's not reading the
entire shift number, just the last few characters. You might want to look
into the csv parser function instead of scanning for a specific number of
characters. It's a little hard to tell from the screenshot if the data
files are consistent enough for this not to matter.

A decoding script doesn't have to do anything when we test it, we're
feeding the data right to it. A routing spec has a lot more moving pieces
to break, so it's not surprising to have more challenges when configuring
them. You're getting closer though!

Andrew

**Reply by marjgh** [reply]
so i solved the decoding issue, and also added the header as other in the source tab as below:
![b](https://github.com/opendcs/opendcs/assets/124637860/16148783-56ab-42d7-aaf7-209c4b4ceef9)
![c](https://github.com/opendcs/opendcs/assets/124637860/43780986-0fc8-40ce-ab82-dd8acbc1ea47)
 also teh log is as below:
![a](https://github.com/opendcs/opendcs/assets/124637860/07a62d6a-423f-4077-8345-ddd1e7897f97)

**Reply by marjgh** [reply]
here is the warning i get:
![g](https://github.com/opendcs/opendcs/assets/124637860/22c07616-76ad-4212-af2f-e184c7b5515b)

**Comment by agilmore2** [comment]
Please find the log for this process to know for sure, as the first line of
the exception trace would tell us for sure where that error occurred.

I'm guessing that you have a working decoding script, but the routing spec
has no way to map the incoming message to a platform ID, which is what it
needs to process the data. There are a couple of ways to get the platform
id, from the file name, file directory, or something in the file contents.
You may also be able to hard code it in the routing spec/decoding pipe.

**Reply by marjgh** [reply]
great, so i have another concern here, this is a GOES station with the following IDs, (webscraping from WSC DataMart, and the GOES ID itself) 
![o](https://github.com/opendcs/opendcs/assets/124637860/3ce9301c-bdde-4a87-bffa-ad86bb64c9e0)

since I wanted to get the shift amounts from WSC webservice, I used another platform, with the same info and same site, but for its ID, instead of 05AD007 (which is a WSC id) I used "=05AD007" as below:
![j](https://github.com/opendcs/opendcs/assets/124637860/225280d2-6644-4543-b7a8-9e9ffc6e9f05)

Is it the reason why the routing is not working? and if so, what do you suggest?

**Comment by agilmore2** [comment]
should be ok, the file name equals the platform id which should work. Need
to tell the routing spec that. Also need to look at the logfile for details
on what's breaking.

**Reply by marjgh** [reply]
how can i set this in routing spec, is it different from what it is now? 
![z](https://github.com/opendcs/opendcs/assets/124637860/fa841557-98b5-4abc-8f86-29686b5df837)
![x](https://github.com/opendcs/opendcs/assets/124637860/cff5ac41-5ed0-4dec-906a-affabdbb1ac8)

also here is the only thing log shows me, I also attached the log file here. I really appreciate your help.
![aa](https://github.com/opendcs/opendcs/assets/124637860/3bbb903e-0d3d-4600-884e-7015e3f18f9d)
[routsched2.log](https://github.com/opendcs/opendcs/files/13925151/routsched2.log)
